plugins/node/opentelemetry-instrumentation-redis-4/src/instrumentation.ts

```diff
@@ -332,6 +338,52 @@ export class RedisInstrumentation extends InstrumentationBase<any> {
     };
   }
 
+  private _getPatchedClientConnect() {
+    const plugin = this;
+    return function connectWrapper(original: Function) {
+      return function patchedConnect(this: any): Promise<void> {
+        const options = this.options;
+
+        const attributes = {
+          [SemanticAttributes.DB_SYSTEM]: DbSystemValues.REDIS,
+          [SemanticAttributes.NET_PEER_NAME]: options?.socket?.host,
+          [SemanticAttributes.NET_PEER_PORT]: options?.socket?.port,
+          [SemanticAttributes.DB_CONNECTION_STRING]: options?.url,
+        };
+
+        const span = plugin.tracer.startSpan(
+          `${RedisInstrumentation.COMPONENT}.connect`,
+          {
+            kind: SpanKind.CLIENT,
+            attributes,
+          }
+        );
+
+        const res = context.with(trace.setSpan(context.active(), span), () => {
+          return original.apply(this);
+        });
+
+        return res
+          .then((result: unknown) => {
+            return new Promise(resolve => {
+              span.end();
+              resolve(result);
+            });
+          })
+          .catch((error: Error) => {
+            return new Promise((_, reject) => {
+              span.setStatus({
+                code: SpanStatusCode.ERROR,
+                message: error.message,
+              });
+              span.end();
+              reject(error);
+            });
+          });
```

Comment:
Actually, when thinking about it again, this might not be equivalent.
If this promise rejects and the user does not catch the error, then without instrumentation the app will crash. If instrumentation adds a `catch` block to the returned promise, then the app will no longer crash which is not desirable.